src/databricks/labs/dqx/check_funcs.py

```diff
@@ -531,6 +561,56 @@ def is_valid_timestamp(column: str | Column, timestamp_format: str | None = None
     )
 
 
+@register_rule("row")
+def is_valid_ipv4_address(column: str | Column) -> Column:
+    """Checks whether the values in the input column have valid IPv4 address formats.
+
+    :param column: column to check; can be a string column name or a column expression
+    :return: Column object for condition
+    """
+    return matches_pattern(column, DQPattern.IPV4_ADDRESS)
+
+
+@register_rule("row")
+def is_ipv4_in_cidr(column: str | Column, cidr_block: str) -> Column:
```

Comment:
For consistency with the other check, can we rename it to `is_ipv4_address_in_cidr` pls